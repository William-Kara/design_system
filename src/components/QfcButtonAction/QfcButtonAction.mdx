import { ArgTypes, Meta } from '@storybook/blocks';
import QfcButtonAction from '@/components/QfcButtonAction/QfcButtonAction.vue';

<Meta title="Elements/ButtonAction" />

## Description

The QfcButtonAction component is a button that can be used for any action. It embeds a notification element that can be toggled.

## Props

The following props are available for the QfcButtonAction component:

<ArgTypes of={QfcButtonAction} />
