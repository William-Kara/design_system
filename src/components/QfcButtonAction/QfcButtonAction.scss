.qfc-btn-action {
  display: flex;
  align-items: center;
  padding: 0.9rem;
  background-color: $background;
  color: $grey-manatee;
  border-radius: $br-xs;
  border: $primary;
  text-align: left;
  cursor: pointer;
  &__content {
    display: flex;
    align-items: center;
    gap: $q-generic-space-3;
    &__icon {
      font-size: $fs-m-2;
    }
    &__label {
      font-size: $fs-xs;
    }
    .icon-is-active {
      color: $yellow;
    }
  }
  &__dot {
    display: block;
    height: 10px;
    width: 10px;
    background-color: $yellow;
    border-radius: 50%;
    position: fixed;
    border: 1px #232323 solid;
    margin-left: 15px;
    margin-bottom: 10px;
  }
  &:hover {
    background-color: $black-hover;
    color: $white;
  }

  .nuxt-link-exact-active {
    .qfc-btn-action {
      color: $primary;
      background-color: transparent;
      &:hover {
        background-color: $black-hover;
      }
    }
  }
}
.button-is-active {
  background-color: $black-lighten;
}

