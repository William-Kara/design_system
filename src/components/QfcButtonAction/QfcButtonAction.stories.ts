import ButtonAction from './QfcButtonAction.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof ButtonAction> = {
  component: ButtonAction,
  title: 'Elements/ButtonAction',
  argTypes: {
    icon: { control: 'text', required: true },
    label: { control: 'text', required: true },
    to: { control: 'text' },
    isNotification: { control: 'boolean' },
    isActive: { control: 'boolean' },
  },
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Share: Story = {
  args: {
    icon: 'share',
    label: 'share',
    to: '#',
    isNotification: false,
    isActive: false,
  },
};
export const Back: Story = {
  args: {
    icon: 'arrow_back',
    label: 'back',
    to: '/docs/elements-buttonaction--docs',
    isNotification: false,
    isActive: false,
  },
};
export const Notifications: Story = {
  args: {
    icon: 'notifications',
    label: 'notifications',
    to: '',
    isNotification: true,
    isActive: false,
  },
};
