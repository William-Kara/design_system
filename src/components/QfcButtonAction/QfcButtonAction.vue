<template>
  <router-link :to="to" custom>
    <button class="qfc-btn-action" :class="{ 'button-is-active': isActive }" role="link">
      <span class="qfc-btn-action__content">
        <span
          class="qfc-btn-action__content__icon material-icons-outlined"
          :class="{ 'icon-is-active': isActive }"
          >{{ icon }}</span
        >
      </span>
      <span v-if="isNotification" class="qfc-btn-action__dot"></span>
    </button>
  </router-link>
</template>

<script lang="ts" setup>
type RelativePath = `/${string}`;
interface Props {
  icon: string;
  label: string;
  to: RelativePath;
  isNotification: boolean;
  isActive: boolean;
}

defineProps<Props>();
</script>

<style lang="scss" scoped src="./QfcButtonAction.scss" />
