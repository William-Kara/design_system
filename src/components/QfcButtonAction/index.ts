import ButtonAction from './QfcButtonAction.vue';

export { ButtonAction };
export default ButtonAction;
