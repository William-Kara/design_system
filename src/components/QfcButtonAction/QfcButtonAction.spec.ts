import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcButtonAction from './QfcButtonAction.vue';

let wrapper: any = null;

const propsValues = {
  icon: 'space_dashboard',
  label: 'Dashboard',
  to: '/dashboard',
  isActive: false,
  isNotification: false,
};

beforeEach(() => {
  wrapper = shallowMount(QfcButtonAction, {
    propsData: propsValues,
  });
});

describe('QfcButtonAction Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
