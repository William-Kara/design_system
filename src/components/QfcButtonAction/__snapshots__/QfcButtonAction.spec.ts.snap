// Vitest Snapshot v1

exports[`QfcButtonAction Component > should render correctly 1`] = `
"<router-link data-v-17646067=\\"\\" to=\\"/dashboard\\" custom=\\"\\"><button data-v-17646067=\\"\\" class=\\"qfc-btn-action\\" role=\\"link\\"><span data-v-17646067=\\"\\" class=\\"qfc-btn-action__content\\"><span data-v-17646067=\\"\\" class=\\"qfc-btn-action__content__icon material-icons-outlined\\">space_dashboard</span></span>
    <!--v-if-->
  </button></router-link>"
`;
